### PR TITLE
Migrate unique token store to use a VirtualMap instead of MerkleMap

### PR DESCRIPTION
**Pull request by rjuang**
**Description**:

This PR migrates the unique token store from a MerkleMap to a VirtualMap, replacing all usages of the MerkleMap with the corresponding VirtualMap method calls. It also adds the required classes and  migration needed to update a node's data holdings from MerkleMap to VirtualMap. 

**Related issue(s)**:

Fixes: https://github.com/hashgraph/hedera-services/issues/2820

**Metrics**
https://github.com/hashgraph/hedera-testing/issues/208

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/state/migration/StateVersions.java [hunk 1]
```diff
@@ -43,9 +43,10 @@ public final class StateVersions {
 	public static final int RELEASE_024X_VERSION = 17;
 	public static final int RELEASE_025X_VERSION = 18;
 	public static final int RELEASE_0260_VERSION = 19;
+	public static final int RELEASE_0270_VERSION = 20;
 
 	public static final int MINIMUM_SUPPORTED_VERSION = RELEASE_024X_VERSION;
-	public static final int CURRENT_VERSION = RELEASE_0260_VERSION;
+	public static final int CURRENT_VERSION = RELEASE_0270_VERSION;
```
Comment (rjuang) [inline hedera-node/src/main/java/com/hedera/services/state/migration/StateVersions.java:49]: NOTE: This needs to be changed to reflect the release version we are targetting.

File: hedera-node/src/main/java/com/hedera/services/state/migration/UniqueTokensMigrator.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import com.swirlds.virtualmap.VirtualMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.hedera.services.utils.MiscUtils.forEach;
+
+public class UniqueTokensMigrator {
+	public static final int TARGET_RELEASE = StateVersions.RELEASE_0270_VERSION;
```
Comment (rjuang) [inline hedera-node/src/main/java/com/hedera/services/state/migration/UniqueTokensMigrator.java:40]: Note: This needs to be updated to reflect the release version we are targetting. 0280 ?

Comment (rjuang) [reply hedera-node/src/main/java/com/hedera/services/state/migration/UniqueTokensMigrator.java]: removed references to version.

**Comment by rjuang** [comment]
### NOTE

This PR to be re-factored into several smaller PRs that will be tracked in #3089 . All PRs will have the title prefix [virtual_nfts]

**Comment by rjuang** [comment]
Converting to draft as there is a requirement for these changes to be feature flag guarded.

**Comment by tinker-michaelj** [comment]
Was completed via a series of smaller PRs.

**Pull request closed without merging**